<?xml version="1.0"?>
<!-- vim: set fenc=utf-8 et sts=4 ft=xml : -->
<launch>

<!-- joy node -->
<node respawn="true" 
      pkg="joy"
      type="joy_node" 
      name="joy_node"
      ns="/io">

    <param name="dev" type="string" value="/dev/input/js0" />
    <param name="deadzone" value="0.05" />
</node>

<node pkg="platform_motion" 
      type="manual_control_node.py" 
      name="manual_control_node"
      ns="/motion">
          
    <!-- Axes -->
    <param name="LINEAR_X" value="1"/>
    <param name="LINEAR_Y" value="0"/>
    <param name="ANGULAR_Z" value="2"/>
    <!-- Buttons -->
    <param name="BUTTON_HOME" value="9"/>
    <param name="BUTTON_GRAB" value="3"/>
    <param name="BUTTON_SERVO" value="1"/>
    <param name="BUTTON_CANCEL" value="2"/>
    <!-- Scaling -->
    <param name="LINEAR_SCALE" value="1" type="double"/>
    <param name="LINEAR_EXP" value="2" type="double"/>
    <param name="ANGULAR_SCALE" value="3.14" type="double"/>
    <param name="ANGULAR_EXP" value="2" type="double"/>
    <!-- Other -->
    <param name="JOY_TIMEOUT" value="0.5" type="double"/>

    <!-- Remaps -->
    <remap from="CAN_select_command_source" to="/motion/CAN/select_command_source"/>
    <remap from="visual_servo_action" to="/processes/visual_servo/servo_action"/>
    <remap from="grab_action" to="/manipulator/grab_action"/>
    <remap from="twist" to="joystick_command"/>
    <remap from="joy" to="/io/joy"/>

</node>

</launch>

